python/lsst/pipe/tasks/postprocess.py

```diff
@@ -1299,3 +1300,75 @@ def writeConfig(self, butler, clobber=False, doBackup=True):
         """No config to write.
         """
         pass
+
+
+class MakeVisitTableConnections(pipeBase.PipelineTaskConnections,
+                                dimensions=("instrument",),
+                                defaultTemplates={}):
+    visitSummaries = pipeBase.connectionTypes.Input(
+        doc="Per-visit consolidated exposure metadata from ConsolidateVisitSummaryTask",
+        name="visitSummary",
+        storageClass="ExposureCatalog",
+        dimensions=("instrument", "visit",),
+        multiple=True,
+        deferLoad=True,
+    )
+    outputCatalog = connectionTypes.Output(
+        doc="Visit metadata table",
+        name="visitTable",
+        storageClass="DataFrame",
+        dimensions=("instrument",)
+    )
+
+
+class MakeVisitTableConfig(pipeBase.PipelineTaskConfig,
+                           pipelineConnections=MakeVisitTableConnections):
+    pass
+
+
+class MakeVisitTableTask(CmdLineTask, pipeBase.PipelineTask):
+    """Produce a `visitTable` from the `visitSummary` exposure catalogs.
+    """
+    _DefaultName = 'makeVisitTable'
+    ConfigClass = MakeVisitTableConfig
+
+    def run(self, visitSummaries):
+        """ Make a table of visit information from the `visitSummary` catalogs
+
+        Parameters
+        ----------
+        visitSummaries : list of `lsst.afw.table.ExposureCatalog`
+            List of exposure catalogs with per-detector summary information.
+
+        Returns
+        -------
+        result : `lsst.pipe.Base.Struct`
+            Results struct with attribute:
+                ``outputCatalog``
+                    Catalog of visit information.
+        """
+        visitEntries = []
+        for visitSummary in visitSummaries:
+            visitSummary = visitSummary.get()
+
+            visitRow = visitSummary[0]
+            visitInfo = visitRow.getVisitInfo()
+            visitEntry = {}
+            visitEntry["visitId"] = visitRow['visit']
+            visitEntry["filterName"] = visitRow['physical_filter']
+            visitEntry["ra"] = np.mean(visitSummary['ra'])
+            visitEntry["decl"] = np.mean(visitSummary['decl'])
+            visitEntry["skyRotation"] = visitInfo.getBoresightRotAngle().asRadians()
+            azimuth, altitude = visitInfo.getBoresightAzAlt()
+            visitEntry["azimuth"] = azimuth.asRadians()
+            visitEntry["altitude"] = altitude.asRadians()
+            visitEntry["zenithDistance"] = visitRow['zenithDistance']
+            visitEntry["airmass"] = visitInfo.getBoresightAirmass()
```

Comment:
Note that the zenith distance in the visit summary table is computed at the center of each ccd.  It is not the boresight zenith distance.  I'm not sure which we want for this table (I personally don't know what "observation mid-point" means technically).  But right now this code snippet will set the zenith distance to the first detector, and the airmass to the boresight.  We need to ensure these are consistent.

Furthermore, "airmass" is a slippery quantity.  There are many ways of computing airmass, and I don't know which one is supposed to be here.  "zenithDistance" is unambiguous, thankfully.